clients/roscpp/src/libros/io.cpp

```diff
@@ -213,18 +289,33 @@ int poll_sockets(socket_pollfd *fds, nfds_t nfds, int timeout) {
 		} else {
 				fds[i].revents = POLLNVAL;
 		}
+                ofds->push_back(fds[i]);
 	}
-	return rc;
+	return ofds;
 #else
-	// use an existing poll implementation
-	int result = poll(fds, nfds, timeout);
-	if ( result < 0 ) {
-		// EINTR means that we got interrupted by a signal, and is not an error
-		if(errno == EINTR) {
-			result = 0;
-		}
-	}
-	return result;
+        UNUSED(nfds);
+        UNUSED(fds);
+        struct epoll_event ev[nfds];
+        boost::shared_ptr<std::vector<socket_pollfd> > ofds;
+
+        int fd_cnt = epoll_wait(epfd, ev, nfds, timeout);
+
+        if (fd_cnt < 0)
+        {
+          ROS_ERROR("Error in epoll_wait! %s", strerror(errno));
```

Comment:
Do we know what conditions would cause this, and if there's something else we need to be doing about them?